Source/AliveLib/Scrab.cpp

```diff
@@ -627,8 +628,483 @@ __int16 Scrab::AI_Patrol_0_4AA630()
+                            if (!field_20_animation.field_4_flags.Get(AnimFlags::eBit18_IsLastFrame))
+                            {
+                                field_108_next_motion = eScrabMotions::M_StandToFeed_35_4AA010;
+                            }
+                        }
+                        result = field_11C_sub_state;
+                        return result;
+                    case 13:
+                        if (Event_Is_Event_In_Range_422C30(kEventAbeOhm, field_B8_xpos, field_BC_ypos, -1))
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        field_106_current_motion = eScrabMotions::M_HowlBegin_26_4A9DA0;
+                        if (field_106_current_motion != eScrabMotions::M_Shriek_30_4A9EA0)
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        if (!field_20_animation.field_4_flags.Get(AnimFlags::eBit18_IsLastFrame))
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+                        return 1;
+                    case 14:
+                        if (field_106_current_motion != eScrabMotions::M_Shriek_30_4A9EA0)
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        if (!field_20_animation.field_4_flags.Get(AnimFlags::eBit18_IsLastFrame))
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+                        return 1;
+                    case 15:
+                        if (field_106_current_motion)
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        ++field_1A2;
+                        if (field_194 == 54)
+                        {
+                            field_108_next_motion = eScrabMotions::M_Shriek_30_4A9EA0;
+                            field_154 = sGnFrame_5C1B84;
+                            return 14;
+                        }
+                        if (field_194 != 53)
+                        {
+                            result = field_11C_sub_state;
+                            return result;
+                        }
+                        field_108_next_motion = eScrabMotions::M_HowlBegin_26_4A9DA0;
+                        result = 13;
+                        break;
+                    case 16:

```

Comment:
you can probably add an enum for these case values since you've figured out some names for them?